### ISSUE TITLE
Samples are too complex

### ISSUE THREAD

**Issue by idish** [android/play-billing-samples#520]
Play billing sample (the trivial drive kotlin one, not sure about the JAVA one as I haven't followed this one), is way too complex to study from and to implement in own app.
There's a massive use of classes, flows, livedata objects and generally very much coding that isn't self-contained. All of that makes it very hard to implement accordingly in own app. The example app is too specific, and is over-complicated in many manners in my opinion.

**Comment by hilgenberg** [comment]
Agreed. This is the absolute worst sample code I've ever seen. All just "look at me, I'm so clever" :-(

**Comment by zhanming95** [comment]
worst sample ever. so hard to even configure.
